tests/TestCase/ORM/TableTest.php

```diff
@@ -1314,6 +1314,30 @@ public function testFindTypedParameters(): void
         $this->assertSame(2, $author->id);
     }
 
+    /**
+     * https://github.com/cakephp/cakephp/issues/18716
+     */
+    public function testChangedFindWithOverlappingArgs(): void
+    {
+        $query = $this->getTableLocator()->get('Authors')
+            ->find('withIdArgument', 2)
+            ->find('custom', id: [1, 2], second: false);
+
+        $this->assertSame([2, 'id' => [1, 2], 'second' => false], $query->getOptions());
```

Comment:
The pain of query options returns.

Reply:
Should we be planning a replacement for 6.0?